cli/src/git_util.rs

```diff
@@ -702,7 +702,7 @@ mod tests {
         };
         // First output is after the initial delay
         assert_snapshot!(update(crate::progress::INITIAL_DELAY - Duration::from_millis(1), 0.1), @"");
-        assert_snapshot!(update(Duration::from_millis(1), 0.10), @"[?25l\r 10% [█▊                ][K");
+        assert_snapshot!(update(Duration::from_millis(1), 0.10), @"\u{1b}[?25l\r 10% [█▊                ]\u{1b}[K");
```

Comment:
Is this supposed to be changed by https://github.com/mitsuhiko/insta/issues/715 ? If that's the case, it's probably better to merge this PR after new insta is released.

Reply:
It's also not very difficult to compile cargo-insta from source. I do it for https://github.com/mitsuhiko/insta/pull/438.

```
cargo build -p cargo-insta --release && cp target/release/cargo-insta ~/.cargo/bin
```

Reply:
> Is this supposed to be changed by [mitsuhiko/insta#715](https://github.com/mitsuhiko/insta/issues/715)?

Yes, this is with https://github.com/mitsuhiko/insta/pull/716 merged.

The current logic is that if there are any bad control characters, then everything will be escaped.
So, because the string contains `\r`, the `\x1b` will be escaped as well.

One could imagine another solution where even if there are unrenderable control characters, only they are escaped, and `\n`, `\t`, `\x1b` are kept as is, but that is not implemented in insta right now.